clang/lib/DPCT/DPCT.cpp

```diff
@@ -614,6 +615,31 @@ int runDPCT(int argc, const char **argv) {
     dpctExit(MigrationSucceeded);
   }
 
+#ifndef _WIN32
+  if (!InterceptBuildCommand.empty()) {
+    SmallString<512> pathToInterceptBuildBinary(DpctInstallPath);
+    llvm::sys::path::append(pathToInterceptBuildBinary, "bin",
+                            "intercept-build");
+    if (!llvm::sys::fs::exists(pathToInterceptBuildBinary)) {
+      DpctLog() << "Error: intercept-build tool not found"
+                << "\n";
+      ShowStatus(MigrationErrorInvalidInstallPath);
+      dpctExit(MigrationErrorInvalidInstallPath);
+    }
+    std::string interceptBuildSystemCall =
+        (pathToInterceptBuildBinary + " " + InterceptBuildCommand).str();
```

Comment:
From the implementation, it only supports the fixed format of "dpct --intercept-build make".
While for the options of intercept-build tool, like:
```
optional arguments:
  -h, --help            show this help message and exit
  --verbose, -v         Enable verbose output from 'intercept-build'. A second, third and fourth flags increases verbosity. (default: 0)
  --cdb <file>          The JSON compilation database. (default: compile_commands.json)
  --parse-build-log <file>
                        Specifies the file path of the build log. (default: None)
  --work-directory <path>
                        Specifies the working directory of the command that generates the build log specified by option --parse-build-log. (default: the directory of build log file specified by option
                        --parse-build-log)
  --linker-entry        Generate linker entry in compilation database if '--linker-entry' option is present. (default: 1)
  --no-linker-entry     Do not generate linker entry in compilation database if '--no-linker-entry' option is present. (default: 0)

advanced options:
  --append              Extend existing compilation database with new entries. Duplicate entries are detected and not present in the final output. The output is not continuously updated, it's done when
                        the build command finished. (default: False)
```
and options for "make" tool:
```
  -b, -m                      Ignored for compatibility.
  -B, --always-make           Unconditionally make all targets.
  -C DIRECTORY, --directory=DIRECTORY
                              Change to DIRECTORY before doing anything.
  -d                          Print lots of debugging information.
  --debug[=FLAGS]             Print various types of debugging information.
  -e, --environment-overrides
                              Environment variables override makefiles.
...
```
the command "dpct --intercept-build" should also support.
For example: "dpct --intercept-build -vvv make", "dpct --intercept-build make -B", "dpct --intercept-build -vvv make -B" should be supported.

Reply:
Fixed by taking direct command line arguments from argv.

Reply:
@daiyaan-ahmed6 “dpct --intercept-build -vvv make” has been supported while options for make still not supported. i.e.: "dpct --intercept-build make -B" still not supported。

Also the note msg should not be emitted during running "dpct --intercept-build -v /usr/bin/make".
···
Chenwei@chenweis-Precision-7920-Tower:~/NVIDIA_CUDA-11.4_Samples/0_Simple/vectorAdd$ dpct --intercept-build -v /usr/bin/make
NOTE: /usr/bin/make is treated as a CUDA file by default. Use the --extra-arg=-xc++ option to treat /usr/bin/make as a C++ file if needed.
NOTE: Could not auto-detect compilation database for file 'make' in '/usr/bin' or any parent directory.
>>> GCC Version is greater or equal to 4.7.0 <<<
make: Nothing to be done for 'all'.

Chenwei@chenweis-Precision-7920-Tower:~/NVIDIA_CUDA-11.4_Samples/0_Simple/vectorAdd$ dpct --intercept-build -v /usr/bin/make -B
Unknown command line argument '-B'. Did you mean '-h'?
dpct exited with code: -16 (Option parsing error, run 'dpct --help' to see supported options and values)

Chenwei@chenweis-Precision-7920-Tower:~/NVIDIA_CUDA-11.4_Samples/0_Simple/vectorAdd$

···

Reply:
"dpct --intercept-build -v make -B" is not being supported as LLVM command line interprets `-` as direct argument to dpct. 

However, "dpct --intercept-build -v make -- -B" works. Do you think implementing it this way is good? I can specify in the description for intercept-build command line option to use `--` to pass positional arguments to build command.

Reply:
Removed the "Note" warning on calling intercept-build

Reply:
> ~"dpct --intercept-build -v make -B" is not being supported as LLVM command line interprets `-` as direct argument to dpct.~
> 
> ~Reference: https://llvm.org/docs/CommandLine.html#specifying-positional-options-with-hyphens~
> 
> ~However, "dpct --intercept-build -v make -- -B" works. Do you think implementing it this way is good? I can specify in the description for intercept-build command line option to use `--` to pass positional arguments to build command.~

Was able to bypass this with llvm::cl::Sink command line option.